web/src/settings_account.ts

```diff
@@ -306,11 +319,16 @@ export function render_user_timezone_dropdown_widget(): void {
                         /*
                             When user selects the option, update the widget to show the selected option.
                             */
-                        const prev = user_settings.timezone;
+                        user_settings.timezone = String(selected);
```

Comment:
I don't think this is needed, this will be handled when event is received for timezone update.